src/main/java/com/exactpro/th2/hand/services/mstore/ProtobufMessageStoreSender.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright 2020-2024 Exactpro (Exactpro Systems Limited)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.exactpro.th2.hand.services.mstore;
+
+import com.exactpro.th2.common.grpc.AnyMessage;
+import com.exactpro.th2.common.grpc.MessageGroup;
+import com.exactpro.th2.common.grpc.MessageGroupBatch;
+import com.exactpro.th2.common.grpc.RawMessage;
+import com.exactpro.th2.common.schema.factory.CommonFactory;
+import com.exactpro.th2.common.schema.message.MessageRouter;
+import com.exactpro.th2.hand.schema.CustomConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class ProtobufMessageStoreSender implements MessageStoreSender<RawMessage> {
+	private static final Logger logger = LoggerFactory.getLogger(ProtobufMessageStoreSender.class);
+	private final MessageRouter<MessageGroupBatch> messageRouterGroupBatch;
+	private final long batchLimit;
+
+
+	public ProtobufMessageStoreSender(CommonFactory factory) {
+		messageRouterGroupBatch = factory.getMessageRouterMessageGroupBatch();
+		CustomConfiguration customConfiguration = factory.getCustomConfiguration(CustomConfiguration.class);
+		this.batchLimit = customConfiguration.getMessageBatchLimit();
+	}
+
+    public void sendMessages(RawMessage messages) {
+		sendMessages(Collections.singleton(messages));
+	}
+
+	public void sendMessages(Collection<RawMessage> messages) {
+		try {
+			sendRawMessages(messages);
+		} catch (Exception e) {
+			logger.error("Cannot store to mstore", e);
+		}
+	}
+
+	@Override
+	public void close() throws Exception {
+		messageRouterGroupBatch.close();
```

Comment:
The router will be closed once we close the common factory. Not sure if we need to close the router here